### DISCUSSION TITLE
babel

### DISCUSSION THREAD

**Discussion by mickdelaney** [egoist/tsup#605] [Q&A]
is the --babel option still available ?
i'm trying to use tsup in a turborepo monorep with react relay.
so i need to use the relay babel plugin in a package 

`"dev": "tsup src/index.tsx --babel --format esm,cjs --watch --dts --external react react-relay relay-runtime",`

https://github.com/mickdelaney/graphql-relay-blog-post/blob/master/client/packages/ships/package.json#L16

**Comment by austinbuckler** [comment]
Seems it was removed at some point after: https://github.com/egoist/tsup/releases/tag/v3.5.0 . Given that it was experimental, I suppose there was a reason. Bummer cause I needed this as well.

**Reply by mickdelaney** [reply]
@austinbuckler i've used the esbuild babel plugin. 

tsup config file below. this will run your 'babel.config.js' 

```

// noinspection JSUnusedGlobalSymbols

import { defineConfig } from 'tsup';
// @ts-ignore
import babel from 'esbuild-plugin-babel';

export default defineConfig(options => {
  return {
    entry: ['src/index.tsx'],
    dts: true,
    esbuildPlugins: [babel()],
    watch: options.watch,
    sourcemap: true,
    format: ['cjs', 'esm'],
    external: ['react', 'react-relay', 'relay-runtime'],
  };
});
```

**Reply by austinbuckler** [reply]
Thanks Mick, I'll give this a shot! Much appreciated 🙏
